gwpy/timeseries/timeseries.py

```diff
@@ -58,6 +58,45 @@ def _fft_length_default(dt):
     return int(max(2, numpy.ceil(2048 * dt.decompose().value)))
 
 
+def _discover_gate_points(ts, thresh, window):
```

Comment:
Is this method a generic 'points above threshold' detector? If so, can we move it into the `Series` class as an instance method and give it an appropriate name?

Reply:
It's a "points above threshold" detector with the automatic clustering that you'd want for gating. If I move it into the `Series` class, would you like me to make the clustering optional?

Reply:
Will replace this with `scipy.signal.find_peaks`